elasticsearch/src/main/java/org/hibernate/search/elasticsearch/schema/impl/DefaultElasticsearchSchemaTranslator.java

```diff
@@ -281,6 +281,15 @@ private void addTypeOptions(PropertyMapping propertyMapping, BridgeDefinedField
 	private void addTypeOptions(PropertyMapping fieldMapping, FacetMetadata facetMetadata) {
 		ExtendedFieldType type;
 
+		if ( facetMetadata.isEncodingAuto() ) {
+			/*
+			 * If the user didn't ask for a specific encoding, just use the same datatype
+			 * as the source field.
+			 */
+			addTypeOptions( fieldMapping, facetMetadata.getSourceField() );
+			return;
+		}
+
```

Comment:
OK, I see how it was the easiest way to fix the issue, considering the current state of the FacetMetadata thing and the fact that dates are mapped to long, which is not very convenient for Es.

But I don't like it very much.

I didn't look at the Lucene part or at which changes would be needed to do that but couldn't we keep the `AUTO` value in the `FacetMetadata` and determine the encoding type in the Lucene specific part, such as what we do here, considering we have the sourceField at hand?

Without looking at the code, I'd say the issue is that for Lucene we probably need to determine the encoding type several times.

Just thinking out loud.